### PR TITLE
use parallel for ci_test

### PR DESCRIPTION
**Pull request by LHT129**
- introduce Makefile new tag: test_asan_parallel
- mark [diskann] & [hnsw] tags run parallel

### CODE REVIEW

File: src/bitset_impl_test.cpp [hunk 1]
```diff
@@ -22,6 +22,10 @@
 
 using namespace roaring;
 
+TEST_CASE("false") {
+    REQUIRE(1 == 0);
```
Comment (jiaweizone) [inline src/bitset_impl_test.cpp]: force failure this task ?

Comment (LHT129) [reply src/bitset_impl_test.cpp]: just for test. remove now

**Review by inabao — APPROVED** [review]

File: test_asan_bg.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  pids+=($!)
+done
+
+for pid in "${pids[@]}"
+do
+  wait $pid
+  exit_codes+=($?)
+done
+
+all_successful=true
```
Comment (jiaweizone) [inline test_asan_bg.sh]: here can direct return `exit_codes` ?

Comment (LHT129) [reply test_asan_bg.sh]: exit codes is an array. we may add some logger to detail which case fail in future

File: test_asan_bg.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
```
Comment (wxyucs) [inline test_asan_bg.sh]: move this file in to `scripts/`

File: test_asan_bg.sh [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+done
+
+./build/tests/functests -d yes ${UT_FILTER} --allow-running-no-tests ${othertag} &
+pids+=($!)
+
+for pid in "${pids[@]}"
+do
+  wait $pid
+  exit_codes+=($?)
+done
```
Comment (wxyucs) [inline test_asan_bg.sh]: what is the behavior here when the second process exit before the first process

Comment (LHT129) [reply test_asan_bg.sh]: no problem, I have checked

File: test_asan_bg.sh (hunk 3 above)
Comment (LHT129) [reply test_asan_bg.sh]: done

**Comment by wxyucs** [comment]
lgtm

**Review by wxyucs — APPROVED** [review]

**Pull request merged**